ValidationLibrary.Rules/HasReadmeRule.cs

```diff
@@ -31,6 +32,7 @@ public class HasReadmeRule : FixableRuleBase<HasReadmeRule>
                         Environment.NewLine +
                         "DO NOT change the name of this Pull Request. Names are used to identify the Pull Requests created by automation.";
         private const string ReadmeFileName = "README.md";
+        private string ReadmeFilePrefix = ReadmeFileName.Substring(0, ReadmeFileName.IndexOf("."));
```

Comment:
1. Avoid `.Substring` in these kind of situations if possible because it's inherently harder to comprehend than just using string interpolation.
2. Move this to `GetReadmeFromBranch` because it's only used there and separate it from `ReadmeFileName`. These things are not related in this case.

Reply:
> 
> 
>     1. Avoid `.Substring` in these kind of situations if possible because it's inherently harder to comprehend than just using string interpolation. Hmm, not sure how to do the same behaviour with string interpolation but yes 👍 
> 
>     2. Move this to `GetReadmeFromBranch` because it's only used there and separate it from `ReadmeFileName`. These things are not related in this case.
I thought that they would be related as changing one would I thought have changed the other but yes if these two variables are not related I'll just de-couple them, maybe it's just easier too. It's still as a const at the top of the rule.

Reply:
This is fine, well done!

But about the relation: In some cases yes, but in case of some older repositories we might want to support other file names ("lueminut.txt", etc.). so even if the value itself is same (and the pull request file name should be one of those supported values), we should not bind them together. There were the same for historical reasons and convience.